app/services/list_publisher.rb

```diff
@@ -0,0 +1,13 @@
+class ListPublisher
+  attr_reader :tag
+
+  def initialize(tag)
+    @tag = tag
+  end
+
+  def perform
+    groups = TopicPresenter.new(tag).build_groups
```

Comment:
Will fix this by extracting a `GroupsPresenter`, since the `build_groups` method doesn't have anything to do with `TagPresenter` anymore.